bin/main.ml

```diff
@@ -0,0 +1,194 @@
+open Eio.Std
+
+let or_die = function
+  | Ok x -> x
+  | Error `Msg m -> failwith m
+
+let pp_timestamp f x =
+  let open Unix in
+  let tm = localtime x in
+  Fmt.pf f "%04d-%02d-%02d %02d:%02d.%02d" (tm.tm_year + 1900) (tm.tm_mon + 1)
+    tm.tm_mday tm.tm_hour tm.tm_min tm.tm_sec
+
+let reporter =
+  let report src level ~over k msgf =
+    let k _ =
+      over ();
+      k ()
+    in
+    let src = Logs.Src.name src in
+    msgf @@ fun ?header ?tags:_ fmt ->
+    Fmt.kpf k Fmt.stderr
+      ("%a %a %a @[" ^^ fmt ^^ "@]@.")
+      pp_timestamp (Unix.gettimeofday ())
+      Fmt.(styled `Magenta string)
+      (Printf.sprintf "%14s" src)
+      Logs_fmt.pp_header (level, header)
+  in
+  { Logs.report }
+
+let setup_log style_renderer level =
+  Fmt_tty.setup_std_outputs ?style_renderer ();
+  Logs.set_level level;
```

Comment:
I thought `Logs` wasn't threadsafe -- are we just careful about using it here, or has that been addressed?

Reply:
Good point. I've added a call to `Logs_threaded.enable ()` to make sure. I wonder if there's some way Logs could do this by itself? e.g. using `Domain.before_first_spawn`.